packages/graphql-transformer-common/src/FunctionResourceIDs.ts

```diff
@@ -9,6 +10,13 @@ export class FunctionResourceIDs {
     return `${FunctionResourceIDs.FunctionDataSourceID(name, region)}Role`;
   }
 
+  static FunctionIAMRoleName(name: string, withEnv: boolean = false): string {
+    if (withEnv) {
+      return `${simplifyName(name).slice(0, 22)}${md5(name).slice(0, 4)}`;
+    }
+    return `${simplifyName(name).slice(0, 32)}${md5(name).slice(0, 4)}`;
```

Comment:
Where is the env name added out here?

Reply:
If the env is added then we are only accounting for the extra space of characters it will take.

Reply:
We need to add the env name - although the function name already has the env-name. If we don't add it, deployment in multi-environments would fail since IAM role names need to be globally unique in an account. 
Please test this change with multiple environments.

Reply:
It is added in the cloudformation template.
An example of the output will look like the following.
```json
"RoleName": {
                    "Fn::If": [
                        "HasEnvironmentParameter",
                        {
                            "Fn::Join": [
                                "-",
                                [
                                    "LongServiceNameFunctiocf96",
                                    {
                                        "Ref": "GetAttGraphQLAPIApiId"
                                    },
                                    {
                                        "Ref": "env"
                                    }
                                ]
                            ]
                        },
                        {
                            "Fn::Join": [
                                "-",
                                [
                                    "LongServiceNameFunctionsProdGetScf96",
                                    {
                                        "Ref": "GetAttGraphQLAPIApiId"
                                    }
                                ]
                            ]
                        }
                    ]
                }
```